ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -859,6 +859,209 @@ CMitsubaLoader::SContext::tex_ass_type CMitsubaLoader::getTexture(SContext& ctx,
 	return std::move(layer);
 }
 
+static bsdf::E_OPCODE BSDFtype2opcode(const CElementBSDF* bsdf)
+{
+	switch (bsdf->type)
+	{
+	case CElementBSDF::Type::DIFFUSE:
+		_IRR_FALLTHROUGH;
+	case CElementBSDF::Type::ROUGHDIFFUSE:
+		return bsdf::OP_DIFFUSE;
+	case CElementBSDF::Type::DIFFUSE_TRANSMITTER:
+		return bsdf::OP_DIFFTRANS;
+	case CElementBSDF::Type::DIELECTRIC:
+		_IRR_FALLTHROUGH;
+	case CElementBSDF::Type::THINDIELECTRIC:
+		_IRR_FALLTHROUGH;
+	case CElementBSDF::Type::ROUGHDIELECTRIC:
+		return bsdf::OP_DIELECTRIC;
+	case CElementBSDF::Type::CONDUCTOR:
+		_IRR_FALLTHROUGH;
+	case CElementBSDF::Type::ROUGHCONDUCTOR:
+		return bsdf::OP_CONDUCTOR;
+	case CElementBSDF::Type::PLASTIC:
+		_IRR_FALLTHROUGH;
+	case CElementBSDF::Type::ROUGHPLASTIC:
+		return bsdf::OP_PLASTIC;
+	case CElementBSDF::Type::COATING:
+		_IRR_FALLTHROUGH;
+	case CElementBSDF::Type::ROUGHCOATING:
+		return bsdf::OP_COATING;
+	case CElementBSDF::Type::BUMPMAP:
+		return bsdf::OP_BUMPMAP;
+	case CElementBSDF::Type::WARD:
+		return bsdf::OP_WARD;
+	case CElementBSDF::Type::BLEND_BSDF:
+	case CElementBSDF::Type::MIXTURE_BSDF:
+	case CElementBSDF::Type::MASK:
```

Comment:
mask we can turn into a bit-option just like twosided

Reply:
mask is getting translated into blend with fully transparent bsdf

Reply:
I think it will be cheaper to just make it into a bitfield option and stick the RGB19E7 weight / bindless texture handle at the very end of the BSDF parameter table.

Reply:
still an issue